### DISCUSSION TITLE
Is there a way to get a partial element of body on click link or fetch?

### DISCUSSION THREAD

**Discussion by ionurboz** [swup/swup#905] [General]
### Is there a way to get a partial element of body on click link or fetch?

When I click on a link on a page that goes to a different page, I want it to append an element I specify inside an element I specify on the main page, instead of changing that page completely.

Example:

```html
--------------- PAGE A ------------
<main>
   <section>...</section>
   <nav>
         <a href="page-b.html" data-target="#main">Get Target</a>
   </nav>
   <div id="example">
        <div id="target"></div> /* (Append here #target's content of Page B) */
   </div>
</main>

--------------- PAGE B ------------
<main>
   <div id="target">Lorem Ipsum</div>
</main>
```

**Comment by daun** [comment]
You can dynamically set the containers to be replaced on each visit. Untested example below. This will **replace** the container, but appending content should be doable by nesting the elements appropriately.

```js
swup.hooks.on('visit:start', (visit) => {
  if (visit.to.url === '/about') {
    visit.containers = ['#target']
  }
})

**Comment by daun** [comment]
For more complex setups and built-in animation support, check out the [Fragment Plugin](https://github.com/swup/fragment-plugin) which allows defining dynamic containers conditionally by url or callback.
